org-eclipse-che-jdt-ls-extension/jdt.ls.extension.core/src/main/java/org/eclipse/che/jdt/ls/extension/core/internal/classpath/ResolveClassPathsHandler.java

```diff
@@ -0,0 +1,153 @@
+import org.eclipse.jdt.ls.core.internal.JDTUtils;
+
+/**
+ * Class for resolving class path and getting location of the output directory for a java project.
+ */
+public class ResolveClassPathsHandler {
+
+  /**
+   * Resolves class path for a java project.
+   *
+   * @param arguments a list contains project URI
+   * @return the class paths entries
+   */
+  public static Set<String> resolveClasspaths(List<Object> arguments) {
+    String projectUri = (String) arguments.get(0);
+
+    IJavaProject javaProject = getJavaProject(projectUri);
+
+    if (javaProject == null) {
+      return emptySet();
+    }
+
+    return getProjectClassPath(javaProject);
+  }
+
+  /**
+   * Gets output location for a java project.
+   *
+   * @param arguments a list contains project URI
+   * @return output location, might returns empty string if something happens
+   */
+  public static String getOutputDirectory(List<Object> arguments) {
+    String projectUri = (String) arguments.get(0);
+
+    IJavaProject javaProject = getJavaProject(projectUri);
+
+    if (javaProject == null) {
+      return "";
+    }
+
+    try {
+      IPath outputLocation = javaProject.getOutputLocation();
+      IPath projectPath = javaProject.getProject().getFullPath();
+      IPath outputLocationWithoutProjectName =
+          outputLocation.removeFirstSegments(projectPath.segmentCount());
+      IFolder outputFolder = javaProject.getProject().getFolder(outputLocationWithoutProjectName);
+      if (outputFolder.exists()) {
+        return outputFolder.getLocation().toOSString();
+      }
+      return "";
+    } catch (JavaModelException e) {
+      return "";
+    }
+  }
+
+  private static IJavaProject getJavaProject(String projectUri) {
+    IWorkspaceRoot root = ResourcesPlugin.getWorkspace().getRoot();
+    IContainer[] containers = root.findContainersForLocationURI(JDTUtils.toURI(projectUri));
+
+    if (containers.length == 0) {
+      return null;
+    }
+
+    IContainer container = containers[0];
+    IProject project = container.getProject();
+    if (!project.exists()) {
+      return null;
+    }
+
+    return JavaCore.create(project);
+  }
+
+  /**
+   * Builds classpath for the java project.
+   *
+   * @param javaProject java project
+   * @return set of resources which are included to the classpath
+   */
+  private static Set<String> getProjectClassPath(IJavaProject javaProject) {
+    try {
+      IClasspathEntry[] resolvedClasspath = javaProject.getResolvedClasspath(false);
+      Set<String> result = new HashSet<>();
+      for (IClasspathEntry classpathEntry : resolvedClasspath) {
+        switch (classpathEntry.getEntryKind()) {
+          case IClasspathEntry.CPE_LIBRARY:
+            IPath path = classpathEntry.getPath();
+            result.add(path.toOSString());
+            break;
+
+          case IClasspathEntry.CPE_SOURCE:
+            IPath outputLocation = classpathEntry.getOutputLocation();
+            if (outputLocation != null) {
```

Comment:
Because `outputLocation.toOSString()` returns  `/{project-artifact-id}/target/classes` not `/{project-folder-name}/target/classes`. For example I create pet-clinic sample with name my-project.
So the result of  `outputLocation.toOSString()` is `/spring-petclinic/target/classes` not `/my-project/target/classes`

Reply:
It looks like the path is an Eclipse workspace path. Remember that the file system location of the projects is mapped into an Eclipse workspace. The right thing to do here is to obain the IResource behind the output location and to get it's location URI. See IContainer

Reply:
reworked